src/sharedQueue.h

```diff
@@ -0,0 +1,142 @@
+/**
+ * \file	serial.h
+ * \author	user66875 (https://stackoverflow.com/a/36763257)
```

Comment:
Are we sure we want to put it like this ? 
And is the whole header license really useful ?

Reply:
Since I used some codes that were already GPL, we better add it